### ISSUE TITLE
[web: canvaskit] resources drawn on the wrong GL context result in bindTexture errors

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#86809]
This happened before ([issue](https://github.com/flutter/flutter/issues/49947)) and was fixed ([PR](https://github.com/flutter/engine/pull/16572)), but it's happening again. I'm suspecting the issue might be because internally CanvasKit doesn't track which context a particular resource is coming from. WebGL, unlike OpenGL, does not allow sharing resources across different contexts, so when we use multiple contexts resources need to be rebound to the currently active context before drawing.

Skia issue: https://bugs.chromium.org/p/skia/issues/detail?id=12255
Internal issue: b/197480079

**Comment by willcalderbank** [comment]
If it helps in anyway, we are seeing this problem when full screening an iframe in a webview.

**Comment by zambetpentru** [comment]
I'm also seeing this and it's causing a lot of problems, is there a reason it is only P3?

**Comment by zambetpentru** [comment]
Hi, is there anything I can do to help diagnose this one? I really really like Flutter and all the progress over the past couple of months, this is the only bug that really concerns me as it makes our forthcoming Production app really unstable. I just tried on the latest master hoping the platform view changes might address the issue and this is what I was able to provoke unfortunately:

Errors Messages:
WebGL: INVALID_OPERATION: bindTexture: object does not belong to this context
WebGL: INVALID_OPERATION: texSubImage2D: no texture bound to target
`
<html>
<body>


Ya | @ | canvaskit.js:240
-- | -- | --
  | $func3758 | @ | canvaskit.wasm:0x1f4c2e
  | $func3742 | @ | canvaskit.wasm:0x1f0191
  | $func6443 | @ | canvaskit.wasm:0x380b07
  | $func2216 | @ | canvaskit.wasm:0xf281c
  | $func4130 | @ | canvaskit.wasm:0x22b3a2
  | $func4131 | @ | canvaskit.wasm:0x22b600
  | $func7714 | @ | canvaskit.wasm:0x3fbf29
  | $func1843 | @ | canvaskit.wasm:0xc215e
  | $func3869 | @ | canvaskit.wasm:0x2075d8
  | $func1399 | @ | canvaskit.wasm:0x8554f
  | $func8922 | @ | canvaskit.wasm:0x4305fc
  | $func1013 | @ | canvaskit.wasm:0x54193
  | $func2721 | @ | canvaskit.wasm:0x1431a6
  | $func2720 | @ | canvaskit.wasm:0x143126
  | $func6573 | @ | canvaskit.wasm:0x3914b9
  | $func9930 | @ | canvaskit.wasm:0x47a6f1
  | $func1203 | @ | canvaskit.wasm:0x68eb9
  | $func2290 | @ | canvaskit.wasm:0xfc642
  | $func8704 | @ | canvaskit.wasm:0x4264f2
  | $func9916 | @ | canvaskit.wasm:0x479bbc
  | $func1496 | @ | canvaskit.wasm:0x8f034
  | $func10556 | @ | canvaskit.wasm:0x49f446
  | $func906 | @ | canvaskit.wasm:0x4b9d7
  | Canvas$drawPicture | @ | VM2320:10
  | w8 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:42827
  | MV | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:13738
  | w8 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31899
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33195
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33124
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33224
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33185
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33185
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33185
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33144
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | i0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33185
  | le | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33082
  | ax4 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33291
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33301
  | aW9 | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9511
  | axy | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33294
  | at_ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:33508
  | ay6 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38329
  | ark | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:73532
  | JJ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:67268
  | JJ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:78105
  | ai3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:67259
  | eval | @ | VM2622:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | RT | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74449
  | Yo | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74434
  | ah7 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74414
  | eval | @ | VM2652:3
  | MC | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9489
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31482
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | JO | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74379
  | agG | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:77907
  | eval | @ | VM2630:3
  | ML | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80484
  | hh | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80700
  | L | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80438
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:93797
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18874
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44302
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44296
  | pt | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:45452
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44699
  | zD | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18993
  | od | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44622
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44666
  | bno | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19025
  | bnI | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19027
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44246
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | childList (async) |   |  
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44253
  | b7t | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19032
  | aUD | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19128
  | mz | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:45460
  | abA | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44629
  | iP | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44627
  | cI | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44288
  | v | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18864
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:21583
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18874
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44302
  | w | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18860
  | aY5 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:21584
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:93784
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18874
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44302
  | w | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18860
  | mV | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:93799
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:93801
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44501
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44277
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12435
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | setTimeout (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | aaK | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44262
  | blI | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18854
  | aZj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18850
  | oK | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:45461
  | cP | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19095
  | h5 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:18931
  | G | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:93775
  | cH | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80772
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80734
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | FV | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80732
  | FV | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80776
  | eL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80731
  | wD | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80617
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80577
  | eL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80959
  | wD | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80617
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80576
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | vB | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80498
  | aqG | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80520
  | JJ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:78104
  | ai3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:67259
  | eval | @ | VM2622:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | RT | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74449
  | Yo | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74434
  | ah7 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74414
  | eval | @ | VM2652:3
  | MC | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9489
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31482
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | tS | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74366
  | aoL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74528
  | eval | @ | VM2826:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74484
  | a1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:43937
  | hM | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:13746
  | Ym | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74426
  | agE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74410
  | eval | @ | VM2650:3
  | a8F | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9492
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31481
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | tS | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74366
  | aoL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74528
  | eval | @ | VM2826:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74484
  | a1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:43937
  | hM | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:13746
  | Ym | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74426
  | agE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74410
  | eval | @ | VM2650:3
  | a8F | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9492
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31481
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | tS | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74366
  | aoL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74528
  | eval | @ | VM2826:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | $2 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74484
  | a1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:43937
  | hM | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:13746
  | Ym | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74426
  | agE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74410
  | eval | @ | VM2650:3
  | a8F | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9492
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31481
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | tS | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74366
  | iN | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74508
  | Ac | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:53167
  | jA | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:53144
  | sh | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:62305
  | aX | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:62240
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80855
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:82029
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:82029
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:82029
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | a0a | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80870
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80997
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | a0a | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80870
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80997
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | a0a | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80870
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80997
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | a0a | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80870
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80997
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80770
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80962
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80770
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80811
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | bU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80791
  | e3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80574
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80738
  | kb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80779
  | jl | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80702
  | vB | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80498
  | aqG | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80520
  | JJ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:78104
  | ai3 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:67259
  | eval | @ | VM2622:3
  | RU | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74445
  | RT | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74449
  | Yo | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74434
  | ah7 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74414
  | eval | @ | VM2652:3
  | MC | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9489
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31482
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | requestAnimationFrame (async) |   |  
  | nrWrapper | @ | VM1437 content:27
  | amE | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48745
  | a_D | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:48744
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:31470
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38324
  | lq | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74389
  | JO | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:74379
  | agG | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:77907
  | eval | @ | VM2630:3
  | ML | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80484
  | hh | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80700
  | L | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:80438
  | wF | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:83619
  | a2A | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:62220
  | Uf | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:62384
  | HC | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:62409
  | eval | @ | VM2913:3
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55282
  | YW | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55568
  | ey | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55572
  | acy | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55259
  | hp | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55228
  | hp | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:55932
  | amL | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54652
  | $0 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54661
  | bno | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19025
  | bnI | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19027
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44246
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27
  | childList (async) |   |  
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:44253
  | b7t | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19032
  | aUD | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19128
  | fC | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:19044
  | UT | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54645
  | arb | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54624
  | hZ | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54702
  | Jz | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54695
  | Jz | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:78084
  | Ry | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54686
  | aus | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54677
  | FW | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54675
  | ai7 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:54671
  | eval | @ | VM2592:3
  | a8F | @ | main.dart.js?MTIvMDg…oyOToxNC43NQ==:9492
  | akY | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38460
  | eval | @ | VM2093:3
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38608
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38597
  | $1 | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:38472
  | bpj | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12436
  | (anonymous) | @ | main.dart.js?MTIvMDg…yOToxNC43NQ==:12444
  | nrWrapper | @ | VM1437 content:27


</body>
</html>`

[.WebGL-000009F60583ED80] GL_INVALID_OPERATION: Invalid combination of format, type and internalFormat.

On the next redraw I get something like this:
![image](https://user-images.githubusercontent.com/81489824/129274618-92125924-3263-4bcf-b51c-c9673f4ad62b.png)

and here, the video has decided not to show up (but will play when I double click on it to make it full screen it plans) or it will render if I change it from a dual page spread to a single page (and thus only 3 objects instead of 5):
![image](https://user-images.githubusercontent.com/81489824/129274858-438c3791-cd91-4002-90c7-330f98578206.png)

Can we get a few more labels such as severe:rendering, stable, has reproductible steps (I asked in related ticket https://github.com/flutter/flutter/issues/87357 if you needed me to try and create a repo)? 

@hterkelsen any ideas or guidance as I see you fixed related https://github.com/flutter/flutter/issues/49947?

**Comment by josh-burton** [comment]
I've been battling a similar issue, where after resizing the window any images flicker and render black with the crazy red content.

I've narrowed it down to using the [pointer_interceptor](https://pub.dev/packages/pointer_interceptor) widget to work around issues with embedded html views intercepting mouse events.

By only adding this widget on a specific screen the issues have disappeared.

**Comment by zambetpentru** [comment]
I won't get back near the computer for a few days but I think there's been a change that made things worse between 2.5.0-6.0.pre.11 and 2.5.0-7.0.pre.53... I will try next week to see if I can quantify it with a sample app.

**Comment by yjbanov** [comment]
Raising priority as more apps are hitting this, seemingly caused by an increased usage of platform view-based plugins, such as the `Link` widget and `package:pointer_interceptor`.

Here's a draft for a Skia-side fix: https://skia-review.googlesource.com/c/skia/+/432136

**Comment by zambetpentru** [comment]
@hterkelsen Greetings, can you share a little background around that change you merged in #28312?

**Comment by Hixie** [comment]
Update from @hterkelsen: _I have a PR which works around the problem for now, which has landed in master. The CanvasKit team have landed a fix to CanvasKit, with a new release coming shortly_

**Comment by hterkelsen** [comment]
Landing https://github.com/flutter/engine/pull/28087 now, which should help with this

**Comment by zambetpentru** [comment]
Brilliant, once it lands I will try my several different scenarios - cheers!

**Comment by zambetpentru** [comment]
I was watching this continue to roll and get stuck with tests and it's finally landed. Just tested now on 2.6.0-1.0.pre.238 with multiple scenarios and it works perfectly, I am delighted and thank you for the fix!

**Comment by tvolkert** [comment]
@yjbanov @hterkelsen two questions:

1. Is this fixed now?
2. Is the issue severe enough that the fix should be cherrypicked into 2.5?

**Comment by yjbanov** [comment]
> 1. Is this fixed now?

Partially.

> 2. Is the issue severe enough that the fix should be cherrypicked into 2.5?

It's pretty severe when it happens, but it depends on the use-case. Strictly speaking, this is not a regression as this has been the case since the beginning. The issue is that our use-case space is growing, and some new use-cases are hitting this issue (for example, adoption of the `Link` widget is creating more pressure on platform views, which induce more WebGL contexts).

@hterkelsen do you think your PR is safe enough for a cherrypick?

**Comment by yjbanov** [comment]
I imagine the https://github.com/flutter/engine/pull/28312 workaround should be pretty safe, although its applicability is limited.

**Comment by tvolkert** [comment]
If you think a cherrypick is warranted and safe, feel free to label this issue with the "cp" label for 2.5.

**Comment by hterkelsen** [comment]
I just added another issue here: https://github.com/flutter/flutter/issues/89986

**Comment by hterkelsen** [comment]
I am working on rolling CanvasKit 30.0 into Flutter web, which fixes this problem. Meanwhile, continue to use the workaround of disabling platform view overlays with `--dart-define=FLUTTER_WEB_MAXIMUM_SURFACES=1`
